File: Day 02/Python/p2-2.py
cube = {
    "blue": 0,
    "green": 0,
    "red": 0
}

somme = 0
f = open("input2.txt", "r")
for x in f:
    cube["blue"] = 0
    cube["green"] = 0
    cube["red"] = 0
    y = x.split(":")[1]
    for z in y.split(";"):
        for a in z.split(","):
            b = a.split()
            if int(b[0]) > cube[b[1]]:
                cube[b[1]] = int(b[0])

    somme += cube["blue"] * cube["green"] * cube["red"]

print(somme)
    

File: Day 02/C++/p2-2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <map>
using namespace std;



int main() {
    map<string, int> cube {
        {"red", 0}, 
        {"green", 0}, 
        {"blue", 0}, 
    };

    fstream monFichier("input2.txt", ios::in);

    long somme = 0;
    string s;
    while (getline(monFichier, s)) {
        cube["red"] = 0;
        cube["green"] = 0;
        cube["blue"] = 0;
        stringstream ligne(s);
        
        string gameNumber;
        string gameValues;
        getline(ligne, gameNumber, ':');
        getline(ligne, gameValues, ':');

        stringstream hands(gameValues);
        string oneHand;
        while (getline(hands, oneHand, ';')) {
            stringstream turn(oneHand);
            string oneGemme;
            while (getline(turn, oneGemme, ',')) {
                stringstream gemme(oneGemme.substr(1));
                string qte;
                string color;
                getline(gemme, qte, ' ');
                getline(gemme, color, ' ');

                if (cube[color] < stoi(qte)) {
                    cube[color] = stoi(qte);
                } 
            }
        }

        somme += (cube["red"] * cube["blue"] * cube["green"]);
    }
    cout << somme << endl;
}